tracer/test/Datadog.Trace.Tests/CallTarget/TaskAsyncContinuationGeneratorTests.cs

```diff
@@ -0,0 +1,339 @@
+// <copyright file="TaskAsyncContinuationGeneratorTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.ClrProfiler.CallTarget.Handlers.Continuations;
+using Xunit;
+
+namespace Datadog.Trace.Tests.CallTarget
+{
+    public class TaskAsyncContinuationGeneratorTests
```

Comment:
Approving this file since this is identical to the updated `TaskContinuationGeneratorTests`, after find/replace `TaskContinuationGeneratorTests` with `TaskAsyncContinuationGeneratorTests` and changing the OnAsyncMethodEnd's to return `Task<TReturn>` instead of `TReturn`